### ISSUE TITLE
Trying to understand the ConfidNet training process

### ISSUE THREAD

**Issue by MacroMayhem** [valeoai/ConfidNet#6]
Hi, I read your corresponding [arxiv](https://arxiv.org/pdf/1910.04851.pdf) article and was trying to understand the learning process as described in the paper. 

> **Learning scheme**. Our complete confidence model, from input image to confidence score, shares its first encoding part (‘ConvNet’ in Fig.2) with the classification model M. The training of ConfidNet starts by fixing entirely M (freezing w) and learning θ using loss (4). 

I understand the main code is implemented in `selfconfid_learner.py`. The baseline model is frozen, bn+dropout is disabled allowing **only** the training of ConfidNet. However, I didn't understand the statement

> In a next step, we can then fine-tune the ConvNet encoder. However, as model M has to remain fixed to compute similar classification predictions, we have now to decouple the feature encoders used for classification and confidence prediction respectively.

I couldn't find the corresponding code which performs the above action. Can you please help me understand what you mean by the above statement?

**Comment by chcorbi** [comment]
Hi @MacroMayhem,

Indeed, first confidence training phase corresponds to the training of ConfidNet layers' parameters &#952;, using model class like `MODEL_selfconfid_classic.py`. 

Then, we further improve performances by allowing to fine-tune the shared ConvNet encoder E. However, to avoid affecting classification predictions, we have to initialize a new encoder E' dedicated to confidence estimation and such that parameters 
w_E' = w_E. This corresponds to the model class like `MODEL_selfconfid_cloning.py`. Thus, this second training phase optimizes 
&#952;' = (w_E', &#952;).
